core/templates/domain/exploration/editable-exploration-backend-api.service.ts

```diff
@@ -15,152 +15,170 @@
 /**
  * @fileoverview Service to send changes to a exploration to the backend.
  */
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-require('domain/exploration/read-only-exploration-backend-api.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-
-require(
-  'pages/exploration-player-page/exploration-player-page.constants.ajs.ts');
-
-angular.module('oppia').factory('EditableExplorationBackendApiService', [
-  '$http', '$q', 'ReadOnlyExplorationBackendApiService',
-  'UrlInterpolationService', 'EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE',
-  'EDITABLE_EXPLORATION_DATA_URL_TEMPLATE',
-  function($http, $q, ReadOnlyExplorationBackendApiService,
-      UrlInterpolationService, EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE,
-      EDITABLE_EXPLORATION_DATA_URL_TEMPLATE) {
-    var _fetchExploration = function(
-        explorationId, applyDraft, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(
-        explorationId, applyDraft);
-      $http.get(editableExplorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
+import cloneDeep from 'lodash/cloneDeep';
+
+import { AppConstants } from 'app.constants';
+import { ReadOnlyExplorationBackendApiService} from
+  'domain/exploration/read-only-exploration-backend-api.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableExplorationBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private readOnlyExplorationBackendApiService:
+      ReadOnlyExplorationBackendApiService,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  private _fetchExploration(
+      explorationId: string, applyDraft: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var editableExplorationDataUrl = this._getExplorationUrl(
```

Comment:
Please use `let` here and in other places in this file.

Reply:
Done